ApartmentBuildingSim.py

```diff
@@ -0,0 +1,141 @@
+import heapq
+import random
+import os
+import jsoneng
+from dotenv import load_dotenv
+from termcolor import *
+
+jdb = jsoneng.JsonDB()
+jdb.create({})
+
+
+
+
+
+
+
+
+
+
+class Char:
+    def __init__(self, name):
+        self.name = name
+        self.state = None
+        self.location = None
+        self.pissCounter = 0
+
+    def __repr__(self):
+        return f"CHAR: name:{self.name} state:{self.state} location:{self.location}"
+        # return f"{colored('CHAR:','red')} name:{self.name} state:{self.state} location:{self.location}"
+
+class WaterDrinker(Char):
+    def createEvents(self, simTime, futureEventList, chars):
+        if not any(self in evt.chars for evt in futureEventList):
+            if self.state == 'idle':
+                heapq.heappush(futureEventList, DrinkWater0(simTime + random.randint(1,1), [self]))
+            
+            if self.state == 'need to piss':
+                heapq.heappush(futureEventList, PissWater0(simTime + random.randint(1,1), [self]))
+                self.pissCounter += 1
+
+
+
+
+
+
+
+
+
+
+
+class Event:
+    def __init__(self, time, chars, name):
+        self.time = time
+        self.name = name
+        self.chars = chars
+
+    def __repr__(self):
+        return f"EVENT: time:{self.time} chars:{self.chars[0].name} name:{self.name}"
+
+    def __lt__(self, other):
+        return self.time < other.time
```

Comment:
Suggestion: define method `handleEvent()` in Event but throw an exception when it is called. This is to communicate to developers that this is an abstract class
![image](https://github.com/youhengzhou/TheSimsPy/assets/48291075/24e5f05f-a104-4c80-8b8e-4dcad12181b6)

```
   def handleEvent(self, futureEventList):
      raise NotImplementedError()
```

Reply:
good suggestion!